pkg/apis/serving/v1alpha1/trainedmodel_webhook.go

```diff
@@ -18,25 +18,37 @@ package v1alpha1
 
 import (
 	"fmt"
+	"github.com/kubeflow/kfserving/pkg/agent/storage"
 	"github.com/kubeflow/kfserving/pkg/utils"
 	"k8s.io/apimachinery/pkg/runtime"
 	"regexp"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"strings"
 )
 
 // regular expressions for validation of isvc name
 const (
-	TmNameFmt                   string = "[a-z]([-a-z0-9]*[a-z0-9])?"
-	InvalidTmNameFormatError           = "the Trained Model \"%s\" is invalid: a Trained Model name must consist of lower case alphanumeric characters or '-', and must start with alphabetical character. (e.g. \"my-name\" or \"abc-123\", regex used for validation is '%s')"
-	InvalidTmMemoryModification        = "the Trained Model \"%s\" memory field is immutable. The memory was \"%s\" but it is updated to \"%s\""
+	CommaSpaceSeparator                 = ", "
+	AlphaNumericDash                    = "[a-z]([-a-z0-9]*[a-z0-9])"
+	TmNameFmt                    string = AlphaNumericDash + "?"
+	StorageUriFmt                string = "((" + AlphaNumericDash + "?)/)*(" + AlphaNumericDash + "?)"
```

Comment:
Can we share the same validation logic as inference service ? I know prefix set might be different but the url validation logic should be same?

Reply:
The inference service only really checks for the prefix so I could extract that logic out and use it for both validators. The validateStorageURI doesn't check for content and I suppose that is because some protocols can contain different characters. I took a look at [GCS Object name requirements](https://cloud.google.com/storage/docs/naming-objects#objectnames) and it seems that vastly differs to [S3 Object key naming](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html). So now I am considering not validating the non-prefix part. What do you think?